x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportCloseJobAction.java

```diff
@@ -331,10 +338,13 @@ public void onResponse(PersistentTasksCustomMetaData.PersistentTask<?> task) {
 
                             @Override
                             public void onFailure(Exception e) {
-                                final int slot = counter.incrementAndGet();
-                                failures.set(slot - 1, e);
-                                if (slot == numberOfJobs) {
-                                    sendResponseOrFailure(request.getJobId(), listener, failures);
+                                if ((e instanceof ResourceNotFoundException &&
+                                    Strings.isAllOrWildcard(new String[]{request.getJobId()})) == false) {
+                                    final int slot = counter.incrementAndGet();
```

Comment:
I think this line needs to come before the `if` statement, otherwise we won't respond to the caller if the race occurs.

Reply:
yeah man, you are absolutely correct. I don't know what I was thinking :). Should have a commit pushed soon to address